stl/inc/algorithm

```diff
@@ -1263,6 +1263,43 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+#ifdef __cpp_lib_concepts
+
+namespace ranges {
+    // ALIAS TEMPLATE copy_result
+    template <class _In, class _Out>
+    using move_result = in_out_result<_In, _Out>;
+
+    class _Move_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out>
+        requires indirectly_movable<_It, _Out> constexpr move_result<_It, _Out> operator()(
+            _It _First, _Se _Last, _Out _Result) const {

```

Comment:
We use a code formatting program (clang-format) to keep the sourcebase all formatted consistently and so we can (mostly) avoid wasting time arguing about how things should be laid out. If you use VS Code with its C++ Extension and "Open Folder" the STL repo (which we recommend) it is even smart enough to run the formatter for you automatically when you save files. This is all extremely handy except for the fact that the clang-format that comes with VS Code generally lags a revision behind the most recent bleeding-edge version that we use. This PR is failing the format validation check (https://github.com/microsoft/STL/pull/888/checks?check_run_id=755206876) so I suspect you are autoformatting with an old clang-format.

I assume that you have LLVM 10 installed - since you haven't mentioned the test runner complaining about being unable to find clang-cl - in which case you simply need to tell the C++ Extension where to find clang-format instead of using its internal version. You can access the appropriate setting at File > Preferences > Settings, make sure to pick "User", drop-down "Extensions", and select "C/C++" then find "Clang_format_path" in there:

![image](https://user-images.githubusercontent.com/456873/84195935-3f0cce00-aa54-11ea-8394-dde8fe66b276.png)

If you installed LLVM in the default location like I did then `c:\Program Files\LLVM\bin\clang-format` is correct for you as well.

Once that's setup, you need to open this file and the test and save them to trigger the autoformatting.

Reply:
Round 2: clang-format doesn't know about concepts and requires clauses yes, so after letting it do its thing we'll want to clean up the mess it makes of these declarations with requires clauses. We'll put `// clang-format off` before the beginning of this function template, and `// clang-format on` after the end of the next function template. Then we want to add an indent to the lines that start with `requires`, break the lines before `constexpr`, and get as close to:
```c++
template <stuff>
    requires things
constexpr return type some_function(args) const {
```
as we can without going over 120 columns.